compiler/dyno/test/resolution/testOperatorOverloads.cpp

```diff
@@ -94,7 +91,145 @@ static void test1() {
   ctx.advanceToNextRevision(false);
 }
 
+// scope issue tests with cast operator
+static void test2() {
+  Context ctx;
+  auto context = &ctx;
+
+  // method and function operator definitions in the same scope should conflict (ambiguous call)
+  QualifiedType qt1 = resolveTypeOfXInit(context,
+    R""""(
+      record R {
+        var field: int;
+      }
+      operator R.:(z: R, type t: int) { return z.field; }
+      operator :(z: R, type t: int) { return z.field; }
+      var myR: R;
+      var x = myR : int;
+    )""""
+  );
+  assert(qt1.kind() == QualifiedType::UNKNOWN);
+  assert(qt1.type() && qt1.type()->isErroneousType());
+  ctx.advanceToNextRevision(false);
+
+  // access to R should implicitly grant access to its method operators
+  QualifiedType qt2 = resolveTypeOfXInit(context,
+    R""""(
+      module M {
+        record R {
+          var field: int;
+        }
+        operator R.:(z: R, type t: int) { return z.field; }
+      }
+
+      import this.M.R;
+      var myR: R;
+      var x = myR : int;
+    )""""
+  );
+  assert(qt2.type() && qt2.type()->isIntType());
+  assert(qt2.kind() == QualifiedType::VAR);
+  ctx.advanceToNextRevision(false);
+}
+
+// various additional tests with other operators
+static void test3() {
+  Context ctx;
+  auto context = &ctx;
+
+  // incorrectly defined unary operator overload
+  QualifiedType qt1 = resolveTypeOfXInit(context,
+    R""""(
+      record R {
+        var field: int;
+        operator !(z: R, type t: int) { return z.field; }
+      }
+      var myR: R;
+      var x = !myR;
+    )""""
+  );
+  assert(qt1.kind() == QualifiedType::UNKNOWN);
+  assert(qt1.type() && qt1.type()->isErroneousType());
+  ctx.advanceToNextRevision(false);
+
+  // correct unary operator overload
+  QualifiedType qt2 = resolveTypeOfXInit(context,
+    R""""(
+      record R {
+        var field: int;
+        operator !(z: R) { return z.field; }
+      }
+      var myR: R;
+      var x = !myR;
+    )""""
+  );
+  assert(qt2.type() && qt2.type()->isIntType());
+  assert(qt2.kind() == QualifiedType::VAR);
+  ctx.advanceToNextRevision(false);
+
+  // overloading more operators
+  QualifiedType qt3 = resolveTypeOfXInit(context,
+    R""""(
+      record R {
+        var field: int;
+        operator +(z: R, t: int) { return z.field; }
+      }
+      var myR: R;
+      var x = myR + 3;
+    )""""
+  );
+  assert(qt3.type() && qt3.type()->isIntType());
+  assert(qt3.kind() == QualifiedType::VAR);
+  ctx.advanceToNextRevision(false);
+
+  QualifiedType qt4 = resolveTypeOfXInit(context,
```

Comment:
Does this test actually exercise any paths that differ from the `+` above?

Reply:
Not as far as this PR's changes, but I thought it would be good to test an operator that is non-arithmetic and yields a different type, just for more regression coverage.